diesel/src/query_builder/bind_collector.rs

```diff
@@ -31,7 +31,7 @@ pub trait BindCollector<'a, DB: TypeMetadata>: Sized {
     ) -> QueryResult<()>
     where
         DB: Backend + HasSqlType<T>,
-        U: ToSql<T, DB> + 'a;
```

Comment:
We need to decide if we consider that a breaking change or not. In the former case we need to find another solution for how to handle unsized bind paramaters like `str` and `[u8]`. This is currently required for [handling `str` and `[u8]` in the custom bind collector implementation](https://gist.github.com/weiznich/19406dde61f5225cd2dc2927a6d33338#file-impl-rs-L613-L722)

Reply:
I think this is only a breaking change for people implementing `BindCollector`, which is:
- People who are implementing a third-party connection, so they are likely relying on `i-opt-into-breaking-changes` anyway
- Very very few people.

-> The cost/ratio seems to be in favor of not considering this a breaking change - unless I'm mistaken on the number of third-party connection implementations out there.

Reply:
I'm aware of 2 third party connection implementations:
* [diesel-oci](https://github.com/GiGainfosystems/diesel-oci), which is essentially maintained by myself so that will not be an issue
* [rsbfclient](https://github.com/fernandobatels/rsfbclient), which uses the ` i-opt-into-breaking-changes` flag anyways.